### ISSUE TITLE
use native ResizeObserver when available

### ISSUE THREAD

**Issue by indirectlylit** [marcj/css-element-queries#184]
Hi there - great library, thank you!

Heads-up that there might be a `ResizeObserver` coming in spec eventually. Would be great if the library would use the native implementation when available.

See:
* https://developers.google.com/web/updates/2016/10/resizeobserver
* https://wicg.github.io/ResizeObserver/
* https://github.com/WICG/ResizeObserver/issues/3

thanks!

**Comment by marcj** [comment]
Yeah, but I guess it will take ages until we have it available.

**Comment by indirectlylit** [comment]
true, [looks like](https://caniuse.com/#search=ResizeObserver) it'll just be chrome for a while
